test/integration/controller/jobs/jobset/jobset_controller_test.go

```diff
@@ -79,204 +80,300 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
+			onDemandFlavor = testing.MakeResourceFlavor("on-demand").Label(instanceKey, "on-demand").Obj()
+			gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).Should(gomega.Succeed())
+			spotFlavor = testing.MakeResourceFlavor("spot").Label(instanceKey, "spot").Obj()
+			gomega.Expect(k8sClient.Create(ctx, spotFlavor)).Should(gomega.Succeed())
+		})
 
-		ginkgo.By("checking the workload is created without queue assigned")
-		createdWorkload := &kueue.Workload{}
-		wlLookupKey := types.NamespacedName{Name: workloadjobset.GetWorkloadNameForJobSet(jobSet.Name, jobSet.UID), Namespace: ns.Name}
-		gomega.Eventually(func() error {
-			return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
-		}, util.Timeout, util.Interval).Should(gomega.Succeed())
-		gomega.Expect(createdWorkload.Spec.QueueName).Should(gomega.Equal(""), "The Workload shouldn't have .spec.queueName set")
-		gomega.Expect(metav1.IsControlledBy(createdWorkload, createdJobSet)).To(gomega.BeTrue(), "The Workload should be owned by the JobSet")
+		ginkgo.AfterEach(func() {
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, spotFlavor, true)
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		})
 
-		ginkgo.By("checking the workload is created with priority and priorityName")
-		gomega.Expect(createdWorkload.Spec.PriorityClassName).Should(gomega.Equal(priorityClassName))
-		gomega.Expect(*createdWorkload.Spec.Priority).Should(gomega.Equal(priorityValue))
+		ginkgo.It("Should reconcile JobSets", func() {
+			ginkgo.By("checking the JobSet gets suspended when created unsuspended")
+			priorityClass := testing.MakePriorityClass(priorityClassName).
+				PriorityValue(priorityValue).Obj()
+			gomega.Expect(k8sClient.Create(ctx, priorityClass)).Should(gomega.Succeed())
 
-		ginkgo.By("checking the workload is updated with queue name when the JobSet does")
-		jobSetQueueName := "test-queue"
-		createdJobSet.Annotations = map[string]string{constants.QueueLabel: jobSetQueueName}
-		gomega.Expect(k8sClient.Update(ctx, createdJobSet)).Should(gomega.Succeed())
-		gomega.Eventually(func() bool {
-			if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
-				return false
-			}
-			return createdWorkload.Spec.QueueName == jobSetQueueName
-		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			jobSet := testingjobset.MakeJobSet(jobSetName, ns.Name).ReplicatedJobs(
+				testingjobset.ReplicatedJobRequirements{
+					Name:        "replicated-job-1",
+					Replicas:    1,
+					Parallelism: 1,
+					Completions: 1,
+				}, testingjobset.ReplicatedJobRequirements{
+					Name:        "replicated-job-2",
+					Replicas:    3,
+					Parallelism: 1,
+					Completions: 1,
+				},
+			).Suspend(false).
+				PriorityClass(priorityClassName).
+				Obj()
+			err := k8sClient.Create(ctx, jobSet)
+			gomega.Expect(err).To(gomega.Succeed())
```

Comment:
```suggestion
			k8sClient.Create(ctx, jobSet).To(gomega.Succeded())
```

Reply:
did you mean:
`gomega.Expect(k8sClient.Create(ctx, jobSet)).To(gomega.Succeed())` ?

Reply:
Oops, that's right :)